reactor-core/src/test/java/reactor/core/scheduler/BoundedElasticSchedulerTest.java

```diff
@@ -680,6 +681,47 @@ public void defaultBoundedElasticConfigurationIsConsistentWithJavadoc() {
 				.isEqualTo(Schedulers.DEFAULT_BOUNDED_ELASTIC_QUEUESIZE);
 	}
 
+	@Test
+	public void raceActiveWorkerDisposeAndDeferredDirectDispose() {
+		BoundedElasticScheduler boundedElasticScheduler = new BoundedElasticScheduler(
+				1, 2, r -> new Thread(r,"raceActiveDeferredDirectDispose"), 3);
+
+		BoundedElasticScheduler.ActiveWorker activeWorker =
+				(BoundedElasticScheduler.ActiveWorker) boundedElasticScheduler.createWorker();
+
+		int maxRounds = 1_000_000;
+		AtomicInteger incrementWon = new AtomicInteger();
+		AtomicInteger rejected = new AtomicInteger();
+		int rounds = 0;
+		while(rounds++ < maxRounds) {
+			try {
+				if (rounds % 10 == 0) { //from time to time reverse the race
+					RaceTestUtils.race(
+							activeWorker::dispose,
+							boundedElasticScheduler.schedule(incrementWon::incrementAndGet)::dispose
```

Comment:
schedule always happens before the race starts due to the method ref on `dispose`. Is it intentionally?

Reply:
good catch, that was at first glance not that bad but in fact we'd better wrap it in a `Runnable`. updated!